### DISCUSSION TITLE
Custom Filter -  Add option to user to select field names dynamically for datasearch

### DISCUSSION THREAD

**Discussion by LastFromtheTop** [appbaseio/reactivesearch#1885] [General]
We want to build custom filter where user can choose field name dynamically and datasearch datafield updates based on field name selection. Two things here.

1. How to fetch all the fieldnames in index show it in dropdown selection
2. Add search options like 'contains', 'does not contain', 'includes', 'does not include'. there are three dropdowns, first one for fieldname selection, second one for search options and third one is dropdown or textinput.

**Comment by siddharthlatest** [comment]
For 1., you can get the mappings of the index and populate the fields dropdown by traversing it.

For 2. input field, you can use a [ReactiveComponent](https://docs.appbase.io/docs/reactivesearch/v3/advanced/reactivecomponent/)  to generate the query based on the fields and the search option value.
